src/Diagnostics.DataProviders/KustoSDKClient.cs

```diff
@@ -95,7 +95,16 @@ public async Task<DataTable> ExecuteQueryAsync(string query, string cluster, str
             {
                 timeTakenStopWatch.Start();
                 var kustoClient = Client(cluster, database);
-                var result = await kustoClient.ExecuteQueryAsync(database, query, clientRequestProperties);
+                var kustoTask = kustoClient.ExecuteQueryAsync(database, query, clientRequestProperties);
+                if (_config.QueryShadowingClusterMapping != null && _config.QueryShadowingClusterMapping.TryGetValue(cluster, out var shadowClusters))
+                {
+                    foreach (string shadowCluster in shadowClusters)
+                    {
+                        var shadowKustoClient = Client(shadowCluster, database);
+                        shadowKustoClient.ExecuteQueryAsync(database, query, clientRequestProperties);
```

Comment:
What would happen to the original query if the query by `shadowKustoClient` fails/times out for some reason?

Reply:
nothing, the task of the async call is not awaited, thus the exception won't be handled. We can check the slow or failed queries in .show query, so I think nothing is need to be logged here at runtimehost side for testing purpose. Let me know if you suggest any changes to this behavior.

Reply:
As long as the original query isn't affected, this approach sounds good to me.

Reply:
Found that a malformed cluster name can cause the exception when calling the `Client`. Added a try catch to swallow the exception.